numpy/core/tests/test_regression.py

```diff
@@ -2412,3 +2412,27 @@ def test_field_access_by_title(self):
             v = str(data[['f1']])
             if HAS_REFCOUNT:
                 assert_(base <= sys.getrefcount(s))
+
+    @pytest.mark.skipif(sys.version_info < (3, 0), reason="Python3 only")
+    def test_datetime_memoryview(self):
+        # gh-11656
+        # Values verified with v1.13.3
+        def as_dict(m):
+            return dict(strides=m.strides, shape=m.shape, itemsize=m.itemsize,
+                        ndim=m.ndim, format=m.format)
+
+        dt1 = np.datetime64('2016-01-01')
+        dt2 = np.datetime64('2017-01-01')
+        expected = {'strides': (1,), 'itemsize': 1, 'ndim': 1,
+                    'shape': (8,), 'format': 'B'}
```

Comment:
One interesting idea would be to set the format to `T{:value:8B  :dtype_m8[N]:(0)B}` so that we can actually round-trip this